RFS/src/main/java/org/opensearch/migrations/bulkload/lucene/LuceneDocumentsReader7.java

```diff
@@ -1,45 +1,35 @@
-package org.opensearch.migrations.bulkload.common;
+package org.opensearch.migrations.bulkload.lucene;
 
 import java.io.IOException;
 import java.nio.file.Path;
-import java.util.List;
 import java.util.concurrent.Callable;
-import java.util.function.Function;
 
-import org.opensearch.migrations.cluster.ClusterSnapshotReader;
+import org.opensearch.migrations.bulkload.common.RfsLuceneDocument;
+import org.opensearch.migrations.bulkload.common.Uid;
 
 import lombok.Lombok;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.lucene.document.Document;
-import org.apache.lucene.index.DirectoryReader;
-import org.apache.lucene.index.IndexCommit;
-import org.apache.lucene.index.IndexReader;
-import org.apache.lucene.index.LeafReaderContext;
-import org.apache.lucene.index.SoftDeletesDirectoryReaderWrapper;
-import org.apache.lucene.store.FSDirectory;
-import org.apache.lucene.util.BytesRef;
 import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 import reactor.core.scheduler.Schedulers;
+import shadow.lucene7.org.apache.lucene.document.Document;
+import shadow.lucene7.org.apache.lucene.index.DirectoryReader;
+import shadow.lucene7.org.apache.lucene.index.IndexReader;
+import shadow.lucene7.org.apache.lucene.index.LeafReaderContext;
+import shadow.lucene7.org.apache.lucene.index.SoftDeletesDirectoryReaderWrapper;
+import shadow.lucene7.org.apache.lucene.store.FSDirectory;
+import shadow.lucene7.org.apache.lucene.util.BytesRef;
 
 @RequiredArgsConstructor
 @Slf4j
-public class LuceneDocumentsReader {
-
-    public static Function<Path, LuceneDocumentsReader> getFactory(ClusterSnapshotReader snapshotReader) {
-        return path -> new LuceneDocumentsReader(
-            path,
-            snapshotReader.getSoftDeletesPossible(),
-            snapshotReader.getSoftDeletesFieldData()
-        );
-    }
+public class LuceneDocumentsReader7 implements LuceneDocumentsReader {
```

Comment:
For later, it we're able to isolate lucene7 vs lucene9 code and we only need one version in the runtime at once, it should be a chip-shot to load the things from the ServiceLoader interface (and not require shadowing at all).

It would also be nice to not have to require as much copied code (e.g. just building/linking for different versions, or injecting different version specific overrides)